src/shared/schemas/todo-schema.ts

```diff
@@ -0,0 +1,36 @@
+import { z } from "zod";
+import { UserSchema } from "./user-schema";
+import { PriorityEnum } from "./priority-schema";
+import { v4 as uuid4 } from "uuid";
+import { STATUS, StatusEnum } from "./status-schema";
+
+export const TodoSchema = z
+  .object({
+    id: z.string().uuid().optional(),
+    name: z
+      .string()
+      .trim()
+      .min(5, "Task Name must be at least 5 characters long.")
+      .max(30, "Task Name must be at most 30 characters long.")
+      .nonempty("Task Name is required."),
+    priority: PriorityEnum,
+    storyPoints: z.coerce
+      .number()
+      .int("Story points must be an integer")
+      .min(1, "Story points must be at least 1")
+      .max(20, "Story points must not exceed 20"),
+    assignee: UserSchema.shape.name,
+    dueDate: z.string().refine((date) => {
+      const parsedDate = new Date(date);
+      return !isNaN(parsedDate.getTime()) && parsedDate >= new Date();
+    }, "Due Date must be a valid future date."),
+    status: StatusEnum.default(STATUS[0]),
+  })
+  .transform((data) => {
+    return {
+      ...data,
+      id: data.id || uuid4(),
+    };
+  });
```

Comment:
This works but probably called more times than expected. This transform function will be called every time that the form is valid and the user makes a valid change. For example, I added a console.log inside the function and I managed to execute it more than 70 times in a few seconds.

<img width="165" alt="Screenshot 2024-12-24 at 8 30 17 AM" src="https://github.com/user-attachments/assets/a13234c7-8a5e-4e68-b437-c1afd966038e" />

A better place to generate the ID would be the onSubmit callback.